tensorly/tenalg/n_mode_product.py

```diff
@@ -59,12 +57,12 @@ def mode_dot(tensor, matrix_or_vector, mode):
             raise ValueError('Can only take n_mode_product with a vector or a matrix.'
                              'Provided array of dimension {} not in [1, 2].'.format(T.ndim(matrix_or_vector)))
 
-        res = T.dot(matrix_or_vector, unfold(tensor, mode))
+        res = T.dot(matrix_or_vector, tl.unfold(tensor, mode))
```

Comment:
Inside the package I tried to just use relative imports. This shouldn't make any difference though -- is there any reason to change? I guess there's also no need to import both `tensorly` and `tensorly.backend`

Since we went with dynamical dispatch, `unfold` and `tl.unfold` should both refer to the correct version.

Reply:
In general, if you import a wrapped function from the `tensorly.contrib.sparse` module (like `from tensorly.contrib.sparse.decomposition import partial_tucker`), then in that wrapped function everything is switched to calling the sparse backend. 

But I think maybe it doesn't matter for `unfold` since it isn't actually a backend function. I think I made this change while trying to fix something that was ultimately caused by something else. Everything should be tested now, so if the tests still pass with this change reverted that's fine.